pinot-query-planner/src/main/java/org/apache/pinot/query/catalog/PinotCatalog.java

```diff
@@ -80,8 +82,20 @@ public Table getTable(String name) {
    */
   @Override
   public Set<String> getTableNames() {
-    return _tableCache.getTableNameMap().keySet().stream().filter(n -> DatabaseUtils.isPartOfDatabase(n, _databaseName))
-        .collect(Collectors.toSet());
+    if (StringUtils.isBlank(_databaseName)) {
+      return _tableCache.getTableNameMap().keySet().stream()
+          .filter(n -> DatabaseUtils.isPartOfDatabase(n, _databaseName))
```

Comment:
This method assumes that tables in the default database will not be explicitly prefixed - is that always true for our table cache here?

Reply:
That's a reasonable doubt, and we can be easily prepared against both scenarios.

https://github.com/apache/pinot/pull/14830/commits/115908e1953cd4690e909307a441006c5ac0bd7f